### ISSUE TITLE
`Register.getTypeFlags()` always returns 0

### ISSUE THREAD

**Issue by LukeSerne** [NationalSecurityAgency/ghidra#5415]
**Describe the bug**
Calling `Register.getTypeFlags()` from a script or the interpreter always returns `0`, even for registers that clearly should have flags set, such as the stack pointer or program counter registers.

**To Reproduce**
Steps to reproduce the behaviour:
1. Open a program in the code browser
2. Open the Python interpreter (Click "Window" > "Python")
3. Enter `currentProgram.getRegister("ESP").getTypeFlags()`. If you didn't open an x86 program, you should choose a name of a register that exists in your language.
4. Observe that the output is `0`.

**Expected behavior**
I expected the output to be nonzero, specifically [`Register.TYPE_SP`](https://ghidra.re/ghidra_docs/api/ghidra/program/model/lang/Register.html#TYPE_SP).

**Environment (please complete the following information):**
 - OS: Manjaro Linux
 - Java Version: 20.0.1
 - Ghidra Version: 10.3
 - Ghidra Origin: locally built

**Additional context**
Note that `Register.getTypeFlags()` seems to return `0` for all registers, not just the stack pointer. However, I chose this register to demonstrate the bug because that register clearly should have a nonzero flags value.

**Comment by LukeSerne** [comment]
@ryanmkurtz, I noticed this issue doesn't have any labels or assignees. Did this issue slip through the cracks?

**Comment by ryanmkurtz** [comment]
No, I have been trying to find the right person to assign it to.  Our best guess is that these flags were defined but never implemented.

**Comment by Veryyes** [comment]
Looking at SlightLanguage.java, Doesn't seem like there's any code that sets the `TYPE_SP` or `TYPE_FP` flags in `Register.typeFlags`. It seems like this should be handled similarly to how the program counter (`TYPE_SP`) flag is set in `SleighLanguage.setProgramCounter()`. Seems like the code looks for the element name "programcounter" in the ".pspec" file.

That said, the x86-64.pspec and x86.pspec files do not having anything that marks up RSP/ESP or RBP/EBP respectively.
